torch/types.py

```diff
@@ -1,7 +1,26 @@
 import torch
-from typing import Union, Sequence
+from typing import Union, Sequence, List, Tuple
+
+import builtins
 
 # Convenience aliases for common composite types that we need
 # to talk about in PyTorch
 
 _TensorOrTensors = Union[torch.Tensor, Sequence[torch.Tensor]]
+
+# In some cases, these basic types are shadowed by corresponding
+# top-level values.  The underscore variants let us refer to these
+# types.  See https://github.com/python/mypy/issues/4146 for why these
+# workarounds is necessary
+_int = builtins.int
```

Comment:
So this is not really a mypy bug, the issue is that `torch.int` exists and means `torch.int32` rather than `builtins.int`. Given that `torch.types` cannot be used in the main `torch` namespace (have to avoid an import cycle), the better approach here would be that there's no shadowing in any other modules, so one can use plain `int` there.